src/components/modals/SupportRequestModal.js

```diff
@@ -0,0 +1,194 @@
+        //${currUrl}
+        uploads: [this.state.attachment.token]
+      },
+      ticket_form_id: 360000669472
+    };
+    await Support.createSupportRequest(ticket);
+    
+    await this.setState(prev => {
+      prev.name = '';
+      prev.type = "question";
+      prev.subject = '';
+      prev.description = '';
+      prev.attachment = '';
+      prev.email = '';
+      return prev;
+    });
+    this.props.onOKRequest('support');;
+  }
+
+  closeHandler() {
+    // this is the method to handle Cancel click
+    // could do some cleaning here
+    // or delegate it to the parent
+    // we need to use it to close the
+    // DO SOMETHING HERE ...
+
+    // and call parent's close handler
+    this.props.onCloseRequest('support');
+  }
+
+  afterOpenHandler() {
+    // call parent's after open handler
+    this.props.onAfterOpen('support');
+
+  }
+
+  nameChangeHandler = (e) => {
+    const nameText = e.target.value;
+    this.setState(prev => {
+      prev.name = nameText;
+      return prev;
+    });
+  };
+
+  typeChangeHandler = (e) => {
+    const typeText = e.target.value;
+    this.setState(prev => {
+      prev.type = typeText;
+      return prev;
+    });
+  };
+
+  subjectChangeHandler = (e) => {
+    const subjectText = e.target.value;
+    this.setState(prev => {
+      prev.subject = subjectText;
+      return prev;
+    });
+  };
+
+  descriptionChangeHandler = (e) => {
+    const descriptionText = e.target.value;
+    this.setState(prev => {
+      prev.description = descriptionText;
+      return prev;
+    });
+  };
+
+  async attachmentChangeHandler(e){
+    const file = e.target.files[0];
+    const attachmentText = await Support.uploadAttachment(file);
+    this.setState(prev => {
+      prev.attachment = attachmentText;
+      return prev;
+    });
+  };
+
+  emailChangeHandler = (e) => {
+    let emailText = e.target.value;
+    this.state.vaild = /.+@.+\.[A-Za-z]+$/.test(emailText) ? true: false; 
+    this.setState(prev => {
+      prev.email = emailText;
+      return prev;
+    });
+  };
+
+  render() {
+
+    return (
+      SupportRequestBaseModal({
+        id: "SupportRequestModal",
+        disableOkBtn: (this.state.name === '' || this.state.email === '' || this.state.subject === '' || this.state.description === '' || this.state.vaild === false),
+        showModal: this.props.showModal,
+        onRequestClose: this.closeHandler,
+        onAfterOpen: this.afterOpenHandler,
+        imgSrc: "/images/icon_add_help.png",
+        color: "common",
+        title: "Contact Us",
+        height: this.state.height,
+        top: this.state.top,
+        action: { label: "Submit", handler: this.OKHandler },
+      },
+
+        [
+          form({ className: "form-horizontal css-form", name: "consentForm", noValidate: true, encType: "multipart/form-data" }, [
+            !this.state.isLogged && div({ className: "form-group first-form-group" }, [
```

Comment:
This construction is throwing me for a loop. Does this mean the div is only rendered if `!this.state.isLogged` resolves to true?

Reply:
I took this construction from the Terra code, I believed it checks if the first conditional is true/false and execute the second part if it's true

Reply:
👍 Makes sense, thanks!